bin/adql_to_lsdb.py

```diff
@@ -0,0 +1,494 @@
+#!/usr/bin/env python3
+
+"""
+Using queryparser, convert ADQL to calls to LSDB.
+
+The template for the target code is:
+import lsdb
+cat = lsdb.open_catalog(
+    "https://lsdb.data/io/hats/gaia_dr3/gaia/",
+    columns=["source_id", "ra", "dec", "phot_g_mean_mag", "phot_variable_flag"],
+    search_filter=lsdb.ConeSearch(ra=270, dec=23, radius_arcsec=3600)
+    )
+q = cat.query(
+    "phot_g_mean_mag < 10 and phot_variable_flag == 'VARIABLE'"
+    )
+return q.head(15)
+"""
+
+import argparse
+import sys
+from queryparser.adql.adqltranslator import ADQLQueryTranslator, FormatListener, SelectQueryListener
+from antlr4 import ParseTreeWalker
+
+class LSDBFormatListener(FormatListener):
+    """
+    Listens to parsing events from a known subset of ADQL, building up the data in these
+    events into data for parameters to LSDB.  These include:
+
+    - Tables to catalogs -> gaia_dr3.gaia -> "https://lsdb.data/io/hats/gaia_dr3/gaia/"
+    - Columns to columns -> SELECT source_id, ra, dec -> columns=["source_id", "ra", "dec"]
+    - CONTAINS(POINT(...), CIRCLE(...))  -> lsdb.ConeSearch(...)
+    - Basic conditions (e.g. phot_g_mean_mag < 10) -> filters= or cat.query(...)
+    - Limits (e.g. TOP 10) -> q.head(limit)
+    """
+    def __init__(self, parser, contexts, limit_contexts):
+        super().__init__(parser, contexts, limit_contexts)
+        self.entities = {
+            'tables': [],
+            'columns': [],
+            'spatial_search': None,
+            'conditions': [],
+            'limits': None
+        }
+        # Track parsing context
+        self._in_contains = False
+        self._current_point = None
+        self._current_circle = None
+        # Track WHERE clause parsing
+        self._in_where = False
+        self._current_conditions = []
+
+    def enterContains(self, ctx):
+        """Enter a CONTAINS clause - set context flag."""
+        self._in_contains = True
+        self._current_point = None
+        self._current_circle = None
+
+    def exitContains(self, ctx):
+        """Exit CONTAINS clause - validate and store spatial search info."""
+        if not self._in_contains:
+            return
+
+        # Validate that we have both POINT and CIRCLE
+        if not self._current_point:
+            raise NotImplementedError("CONTAINS clause must include a POINT")
```

Comment:
Is it NotImplementedError and not Value Error here to allude to possible future support for searches such as BoxSearch, PolygonSearch, etc, that we just haven't implemented yet?